aito/sdk/aito_client.py

```diff
@@ -0,0 +1,565 @@
+        :param entries: list of the table entries
+        :type entries: List[Dict]
+        :param batch_size: the batch size, defaults to 1000
+        :type batch_size: int, optional
+        :param optimize_on_finished: `optimize https://aito.ai/docs/api/#post-api-v1-data-table-optimize`__ the table on finished, default to True
+        :type optimize_on_finished: bool
+        """
+        LOG.debug(f'uploading {len(entries)} entries to table `{table_name}` with batch size of {batch_size}...')
+        begin_idx = 0
+        populated = 0
+        while begin_idx < len(entries):
+            last_idx = begin_idx + batch_size
+            if last_idx > len(entries):
+                last_idx = len(entries)
+            entries_batch = entries[begin_idx:last_idx]
+            try:
+                LOG.debug(f'uploading batch {begin_idx}:{last_idx}...')
+                self.upload_entries(table_name, entries_batch)
+                populated += len(entries_batch)
+                LOG.info(f'uploaded batch {begin_idx}:{last_idx}')
+            except Exception as e:
+                LOG.error(f'batch {begin_idx}:{last_idx} failed: {e}')
+            begin_idx = last_idx
+
+        LOG.info(f'uploaded {populated}/{len(entries)} entries to table `{table_name}`')
+
+        if optimize_on_finished:
+            self.optimize_table(table_name)
+
+    def upload_binary_file(
+            self, table_name: str, binary_file: BinaryIO, optimize_on_finished: bool = True, polling_time: int = 10
+    ):
+        """upload a binary file object to a table `API doc <https://aito.ai/docs/api/#post-api-v1-data-table-file>`__
+
+        :param table_name: the name of the table
+        :type table_name: str
+        :param binary_file: binary file object
+        :type binary_file: BinaryIO
+        :param optimize_on_finished: `optimize https://aito.ai/docs/api/#post-api-v1-data-table-optimize`__ the table on finished, default to True
+        :type optimize_on_finished: bool
+        :param polling_time: polling wait time
+        :type polling_time: int
+        :raises RequestError: an error occurred during the upload of the file to S3
+        """
+        LOG.debug(f'uploading file object to table `{table_name}`...')
+        LOG.debug('initiating file upload...')
+        r = self.request('POST', f"/api/v1/data/{table_name}/file")
+        LOG.info('file upload initiated')
+        LOG.debug('getting session id and upload url...')
+        upload_session_id = r['id']
+        session_end_point = f'/api/v1/data/{table_name}/file/{upload_session_id}'
+        s3_url = r['url']
+        upload_req_method = r['method']
+        LOG.info('got session id and upload url')
+        LOG.debug('uploading file file to s3...')
+        try:
+            r = requests.request(upload_req_method, s3_url, data=binary_file)
+            r.raise_for_status()
+        except Exception as e:
+            raise BaseError(f'fail to upload file to S3: {e}')
```

Comment:
```suggestion
            raise BaseError(f'failed to upload file to S3: {e}')
```

And as said previously, think if more specific error could be used in the exception and catch everything under Exception.